### PR TITLE
Update to the latest ARM stuff from msun.

### PR DESCRIPTION
**Pull request by ViralBShah**
Many more tests pass now, but has to be compiled with -O0 to avoid
segfault in the rounding tests.

### CODE REVIEW

File: Make.inc [hunk 1]
```diff
@@ -43,7 +43,7 @@ ifeq ($(ARCH),mingw32)
 $(error "the mingw32 compiler you are using fails the openblas testsuite. please see the Julia README.windows.md document for a replacement")
 endif
 
-CFLAGS_add += -std=c99 -Wall -O3 -I$(OPENLIBM_HOME) -I$(OPENLIBM_HOME)/include -I$(OPENLIBM_HOME)/ld80 -I$(OPENLIBM_HOME)/$(ARCH) -I$(OPENLIBM_HOME)/src -DASSEMBLER -D__BSD_VISIBLE -Wno-implicit-function-declaration
+CFLAGS_add += -std=c99 -Wall -O1 -Wno-implicit-function-declaration -DASSEMBLER -D__BSD_VISIBLE -I$(OPENLIBM_HOME) -I$(OPENLIBM_HOME)/include -I$(OPENLIBM_HOME)/ld80 -I$(OPENLIBM_HOME)/$(ARCH) -I$(OPENLIBM_HOME)/src
```
Comment (nalimilan) [inline Make.inc]: Why `-O1`? Is that required for ARM? Doesn't look like a good default for a performance-sensitive library.

Comment (ViralBShah) [inline Make.inc]: Actually it needs to be -O0. Tests are crashing at higher levels of optimization.

**Comment by ViralBShah** [comment]
According to http://svnweb.freebsd.org/base/head/lib/msun/arm/Makefile.inc?view=log one has to do some strange things with the soft float ABI even for VFP, which is hard float.

**Comment by ViralBShah** [comment]
With this patch, we go from all exception tests failing and a crash in the test suite, to passing with only 3 minor errors.

File: Make.inc (hunk 1 above)
Comment (tkelman) [inline Make.inc]: can we make the change in optimization level arm-only?

Comment (ViralBShah) [inline Make.inc]: Yes, we could - but it will be kind of pointless. Better to just use the system libm till this is fixed.

Comment (nalimilan) [inline Make.inc]: I don't get it: do you want to merge this? This code is in openlibm, so people using this code are not using the system libm. Better keep `-O2` for i386/x86_64, where openlibm is used.

Comment (ViralBShah) [inline Make.inc]: I want to try a few more things, and then merge this with -O0 for ARM if I can't fix the segfault.

**Comment by ViralBShah** [comment]
The arm branch needs more work. I have rebased, but it fails to find `sys/cdefs.h`.

**Comment by ViralBShah** [comment]
This is folded into the `vs/msun` branch.

**Pull request closed without merging**